frontend/javascript/products.js

```diff
@@ -84,14 +100,25 @@ function appendProducts(products) {
     // for the presence of the template element's content attribute.
     if (doesBrowserSupportTemplete) {
         products.forEach((product) => {
-            const templateClone = createTemplate();
+            const templateClone = createTemplate('#trending-product-template');
 
             createProductImage(templateClone);
 
             createProductLabel(templateClone, product);
 
+            templateClone
+                .querySelector('.trending-div-product-addcart')
+                .setAttribute('data-test', product.id);
```

Comment:
A detail, but I would choose a different attribute name than `data-test` here, since test usually indicates that it's being used for targeting elements in unit/UI tests. 

For example `data-product-id` could be suitable perhaps? Up to you if you'd like to change it, obviously it works this way and this is just an exercise. 🙂

Reply:
attribute name has been changed